pkg/apis/kops/cluster.go

```diff
@@ -488,7 +488,7 @@ func (z *ClusterZoneSpec) assignCIDR(c *Cluster) error {
 	// We assume a maximum of 8 subnets per network
 	// TODO: Does this make sense on GCE?
 	// TODO: Should we limit this to say 1000 IPs per subnet? (any reason to?)
-	index = index % 8
+	index = index % 8 - 1
 	networkLength += 3
```

Comment:
I tested this for various network-cidr option values:
10.0.0.0/22
```
I1118 07:22:31.466899   55710 cluster.go:503] Assigned CIDR 10.0.1.0/25 to zone us-east-1c
I1118 07:22:31.466908   55710 cluster.go:514] Assigned Private CIDR 10.0.2.128/25 to zone us-east-1c
I1118 07:22:31.466913   55710 cluster.go:503] Assigned CIDR 10.0.1.128/25 to zone us-east-1d
I1118 07:22:31.466917   55710 cluster.go:514] Assigned Private CIDR 10.0.3.0/25 to zone us-east-1d
I1118 07:22:31.466922   55710 cluster.go:503] Assigned CIDR 10.0.2.0/25 to zone us-east-1e
I1118 07:22:31.466925   55710 cluster.go:514] Assigned Private CIDR 10.0.3.128/25 to zone us-east-1e
```
10.0.0.0/23
```
I1121 13:47:13.742974    9973 cluster.go:503] Assigned CIDR 10.0.0.128/26 to zone us-east-1c
I1121 13:47:13.743005    9973 cluster.go:514] Assigned Private CIDR 10.0.1.64/26 to zone us-east-1c
I1121 13:47:13.743016    9973 cluster.go:503] Assigned CIDR 10.0.0.192/26 to zone us-east-1d
I1121 13:47:13.743040    9973 cluster.go:514] Assigned Private CIDR 10.0.1.128/26 to zone us-east-1d
I1121 13:47:13.743050    9973 cluster.go:503] Assigned CIDR 10.0.1.0/26 to zone us-east-1e
I1121 13:47:13.743058    9973 cluster.go:514] Assigned Private CIDR 10.0.1.192/26 to zone us-east-1e
```
10.0.0.0/24
```
I1121 13:50:34.825381    9995 cluster.go:503] Assigned CIDR 10.0.0.64/27 to zone us-east-1c
I1121 13:50:34.825404    9995 cluster.go:514] Assigned Private CIDR 10.0.0.160/27 to zone us-east-1c
I1121 13:50:34.825415    9995 cluster.go:503] Assigned CIDR 10.0.0.96/27 to zone us-east-1d
I1121 13:50:34.825423    9995 cluster.go:514] Assigned Private CIDR 10.0.0.192/27 to zone us-east-1d
I1121 13:50:34.825432    9995 cluster.go:503] Assigned CIDR 10.0.0.128/27 to zone us-east-1e
I1121 13:50:34.825440    9995 cluster.go:514] Assigned Private CIDR 10.0.0.224/27 to zone us-east-1e
```
10.0.0.0/25
```
I1121 13:52:12.468827   10006 cluster.go:503] Assigned CIDR 10.0.0.32/28 to zone us-east-1c
I1121 13:52:12.468857   10006 cluster.go:514] Assigned Private CIDR 10.0.0.80/28 to zone us-east-1c
I1121 13:52:12.468868   10006 cluster.go:503] Assigned CIDR 10.0.0.48/28 to zone us-east-1d
I1121 13:52:12.468875   10006 cluster.go:514] Assigned Private CIDR 10.0.0.96/28 to zone us-east-1d
I1121 13:52:12.468884   10006 cluster.go:503] Assigned CIDR 10.0.0.64/28 to zone us-east-1e
I1121 13:52:12.468892   10006 cluster.go:514] Assigned Private CIDR 10.0.0.112/28 to zone us-east-1e
```

Every time code generated subnets within vpc range, while without this change we overlapped.
I have to admit, this was an empirically calculated change. I do not have a mathematical proof of this, but I ran enough tests to see that it needs to be done for private topology functionality to be usable.